CodeU_Assignement_5/src/codeu_assignement_5/kemUnknownLanguage.java

```diff
@@ -0,0 +1,177 @@
+package codeu_assignement_5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Karine
+ */
+public class kemUnknownLanguage 
+{
+    /*
+     * Input: a dictionary (a list of words in lexicographic order) of all words in an unknown/invented language, 
+    *  Output: alphabet (an ordered list of characters) of that language, letters we cannot decide are appended in the end.
+    */
+    public List<Character> getAlphabeit(String[] alienWords)
+    {
+        List<Character> dict = null;
+        if ((alienWords != null) && (alienWords.length > 0))
+        {
+            /* Get the current order by prefix size, starting from 0 */
+            dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+            /* A,R,C from the example! */
+            
+            /* Add from prefix while there are letters after first letter of prefix */
+            while (!addToDictByPrefix(alienWords, dict)){} 
+            
+            /* Add all the chars we cannot decide in the end*/
+            for (String w : alienWords) 
+            {
+                for (Character c: w.toCharArray())
+                {
+                    if (!dict.contains(c))
+                    {
+                        dict.add(c);
+                    }
+                }
+            }
+        }
+        
+        return dict;
+    }
+    
+    /*
+     * Inputs: All words with the same prefix (all the first prefixSize chars are the same!)
+     * Output: Per common prefix, gets the order of the chars
+     *
+     * Note: Assume the input is valid here!
+    */
+    private List<Character> getBasicAlphaBeit(String[] words, int prefixSize)
+    {
+        List<Character> ret = new ArrayList(); 
+        if (words == null)
+        {
+            return ret;
+        }
+        
+        for (String w : words) 
+        {
+            if (!ret.contains(w.charAt(prefixSize)))
+            {
+                ret.add(w.charAt(prefixSize));
+            }
+        }   
+        
+        return ret;
+    }
+    
+    /*
+     * Gives the prefix of a char that isn't in the dictionary
+    */
+    private int getSizeOfPrefix(String w, List<Character> dict)
+    {
+        if ((w == null) || (w.length() == 0) || (dict == null) || (dict.isEmpty()))
+        {
+            return 0;
+        }

```

Comment:
This is a private method so you control what gets passed - I would be tempted to get rid of the parameter check